starport/services/scaffolder/component.go

```diff
@@ -76,8 +60,28 @@ func isMsgServerDefined(appPath, moduleName string) (bool, error) {
 	return true, err
 }
 
-// isForbiddenComponentName returns true if the name is forbidden as a component name
-func isForbiddenComponentName(name string) bool {
+// checkComponentValidity performs various checks common to all components to verify if it can be scaffolded
+func checkComponentValidity(appPath, moduleName, compName string) error {
+	ok, err := moduleExists(appPath, moduleName)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return fmt.Errorf("the module %s doesn't exist", moduleName)
+	}
+
+	// Ensure the name is valid, otherwise it would generate an incorrect code
+	if err := checkForbiddenComponentName(compName); err != nil {
+		return fmt.Errorf("%s can't be used as a component name: %s", compName, err.Error())
```

Comment:
We would also combine two errors in `checkForbiddenComponentName` because `checkGoReservedWord` is more general and used for field name as well. Also `checkForbiddenComponentName` only check if the name is valid while `checkComponentValidity` check if the component can be scaffolded and so precise why it cannot be scaffolded